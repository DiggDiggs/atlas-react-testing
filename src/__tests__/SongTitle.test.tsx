import React from "react";
import { render } from "@testing-library/react";
import SongTitle from "../components/SongTitle";

describe("SongTitle Component", () => {
  it("should match the snapshot", () => {
    const mockProps = {
      title: "Test Song Title",
      artist: "Test Artist Name",
    };

    const { asFragment } = render(<SongTitle {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render with different title and artist", () => {
    const mockProps = {
      title: "Another Song Title",
      artist: "Another Artist Name",
    };

    const { asFragment } = render(<SongTitle {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
