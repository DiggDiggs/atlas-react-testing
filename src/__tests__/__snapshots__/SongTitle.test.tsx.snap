// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle Component > should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="mb-2 text-2xl font-bold leading-8 text-steel-blue-700"
    >
      Test Song Title
    </h2>
    <p
      class="mb-4 text-steel-blue-700"
    >
      Test Artist Name
    </p>
  </div>
</DocumentFragment>
`;

exports[`SongTitle Component > should render with different title and artist 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="mb-2 text-2xl font-bold leading-8 text-steel-blue-700"
    >
      Another Song Title
    </h2>
    <p
      class="mb-4 text-steel-blue-700"
    >
      Another Artist Name
    </p>
  </div>
</DocumentFragment>
`;

exports[`SongTitle Component should match the snapshot 1`] = `
<div>
  <h2
    class="mb-2 text-2xl font-bold leading-8 text-steel-blue-700"
  >
    Test Song Title
  </h2>
  <p
    class="mb-4 text-steel-blue-700"
  >
    Test Artist Name
  </p>
</div>
`;
